### ISSUE TITLE
Doesn't handle references inside generics (e.g. Option)

### ISSUE THREAD

**Issue by puckipedia** [dcchut/async-recursion#34]
```rust
#[async_recursion::async_recursion]
async fn meow(n : Option<&str>) -> () {
    meow(n).await
}
```

expands to (cleaned):
```rust
#[must_use]
fn meow(
    n: Option<&str>,
) -> Pin<Box<dyn Future<Output = ()> + Send>> {
    Box::pin(async move { meow(n).await })
}
```

**Comment by dcchut** [comment]
nice catch - I'll have a look at improving the way lifetimes are extracted from the function's signature.

**Comment by dcchut** [comment]
Released in v1.1.0.
